FrontEnd/src/components/ProfileDetail/DetailedInfo/PhoneEmail.jsx

```diff
@@ -1,41 +1,44 @@
 import { useState } from 'react';
 import useSWR from 'swr';
+import { useUser } from '../../../hooks';
 import { PropTypes } from 'prop-types';
 import classes from './PhoneEmail.module.css';
 
-function PhoneEmail ({ isAuthorized, profileId }) {
+function PhoneEmail ({ profileId, personId }) {
     const [isPhoneShown, setPhoneShown] = useState(false);
     const [isEmailShown, setEmailShown] = useState(false);
     const authToken = localStorage.getItem('Token');
+    const { user } = useUser();
 
-    // TODO: change the logic of getting user.id when PR with hooks will be merged
-
-    const { data: userData } = useSWR(
-        authToken ? `${process.env.REACT_APP_BASE_API_URL}/api/auth/users/me/` : null,
-        url =>
-            fetch(url, {
-                method: 'GET',
-                headers: {
-                    'Authorization': `Token ${authToken}`,
-                },
-            }).then(res => res.json()),
-    );
-
-    const { data: profileData} = useSWR(
-        authToken ? `${process.env.REACT_APP_BASE_API_URL}/api/profiles/${profileId}?with_contacts=True` : null,
-        url =>
-            fetch(url, {
-                method: 'GET',
-                headers: {
-                    'Authorization': `Token ${authToken}`,
-                },
-            }).then(res => res.json()),
+    const { data: profileData } = useSWR(
+      `${process.env.REACT_APP_BASE_API_URL}/api/profiles/${profileId}?with_contacts=True`,
+      (url) =>
+        fetch(url, {
+          method: 'GET',
+          headers: {
+            Authorization: `Token ${authToken}`,
+          },
+        })
+          .then((res) => {
+            if (!res.ok && res.status === 401) {
+              const error = new Error('Unauthorized user');
+              error.info = res.json();
+              error.status = res.status;
+              throw error;
+            }
+            return res.json();
+          })
+          .catch((error) => {
+            if (error.status === 401) {
```

Comment:
Do we want to log error only if 401 occur?

Reply:
This component will be updated as soon as appropriate changes on back-end will be made. And I assume that this code will be removed.